### PR TITLE
Honor roster system for ssh orchestrations

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Honor roster file when passed to orchestration state.

### What issues does this PR fix or reference?

#51158

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

File: salt/states/saltmod.py [hunk 1]
```diff
@@ -284,6 +284,8 @@ def state(name,
 
     cmd_kw['tgt_type'] = tgt_type
     cmd_kw['ssh'] = ssh
+    if 'roster' in kwargs:
+        cmd_kw['roster_file'] = kwargs['roster']
```
Comment (AndreiPashkin) [inline salt/states/saltmod.py]: `roster` argument refers to roster-system, not roster-file. "Roster-file" is relevant within to "flat" roster system, but Salt allows other roster-systems to be used including custom ones.

**Review by AndreiPashkin — CHANGES_REQUESTED** [review]

File: salt/states/saltmod.py (hunk 1 above)
Comment (dwoz) [reply salt/states/saltmod.py]: The documentation says that `roster` for the orchestration state refers to a roster file. This PR is just to make the state honor what the documentation says. Passing the `roster_file` keyword arg to `saltutil.cmd` uses the default `roster` value of `flat`

Comment (AndreiPashkin) [reply salt/states/saltmod.py]: > The documentation says that roster for the orchestration state refers to a roster file.

Not it's not:

> roster
> In the event of using salt-ssh, a **roster system** can be set

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.state
Notice - roster **system**, not roster **file**

Comment (dwoz) [reply salt/states/saltmod.py]: Okay, thanks for pointing that out. I'll look at this again.

Comment (dwoz) [reply salt/states/saltmod.py]: @AndreiPashkin Can you reply to #51158 with the expected behavior for this option? I will try to re-implement this based on your feedback in the ticket.

Comment (dwoz) [reply salt/states/saltmod.py]: @AndreiPashkin Are we good here?

**Review by AndreiPashkin — APPROVED** [review]

**Pull request merged**